subsys/bluetooth/host/hci_core.c

```diff
@@ -1032,10 +1032,20 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	struct bt_conn *conn;
 	int err;
 
-	BT_DBG("status 0x%02x handle %u role %u %s", evt->status, handle,
-	       evt->role, bt_addr_le_str(&evt->peer_addr));
+	BT_DBG("status 0x%02x handle %u role %u adr:%s rpa:%s local_rpa:%s",
+			evt->status, handle, evt->role, bt_addr_le_str(&evt->peer_addr),
+			bt_addr_str(&evt->peer_rpa), bt_addr_str(&evt->local_rpa));
 
 #if defined(CONFIG_BT_SMP)
+	if (!bt_addr_cmp(&evt->peer_addr.a, BT_ADDR_NONE)) {
```

Comment:
This extra code should be build-time conditional to the HCI driver that exhibits this issue. You should also add a code comment here explaining it (basically the same as you have in the commit message). We could also entertain the thought of introducing driver quirks (same as Linux HCI drivers have), but I think a Kconfig dependency is enough for now.

Alternatively, could you do the fixup directly inside the HCI driver, to avoid polluting the HW-agnostic host-side code with this kind of workarounds?

Reply:
I think that putting the code in ipm_stm32wb.c might be the best option. That would leave the least amount of clutter I think. Especially in case somebody wants to use the BT_ADDR_NONE address with an IRK.